Prose/ProseLib/Sources/NotificationsClient/NotificationsClient+Live.swift

```diff
@@ -0,0 +1,76 @@
+import Combine
+import ComposableArchitecture
+import Foundation
+import Toolbox
+import UserNotifications
+
+public extension NotificationsClient {
+    static let live: Self = {
+        let permissionSubject = CurrentValueSubject<NotificationPermission, Never>(.notDetermined)
+
+        func refreshNotificationPermission() {
+            UNUserNotificationCenter.current().getNotificationSettings { settings in
+                permissionSubject.send(.init(authorizationStatus: settings.authorizationStatus))
+            }
+        }
+
+        refreshNotificationPermission()
```

Comment:
This is smart. I just wonder if it could be delayed without adding too much complexity 🤔

Reply:
Delayed why?

Reply:
To reduce the app startup time?

Reply:
Static vars, like lazy, are only evaluated when called. I also wouldn't think that `getNotificationSettings` is expensive as it even uses a callback to return the result.